kola/tests/misc/nvidia.go

```diff
@@ -3,50 +3,239 @@ package misc
+    contents:
+      source: https://github.com/flatcar/sysext-bakery/releases/download/latest/nvidia_runtime-v1.16.2-{{ .ARCH_SUFFIX }}.raw
+  links:
+  - path: /etc/extensions/kubernetes.raw
+    target: /opt/extensions/kubernetes-v1.30.4-{{ .ARCH_SUFFIX }}.raw
+    hard: false
+  - path: /etc/extensions/nvidia_runtime.raw
+    target: /opt/extensions/nvidia_runtime-v1.16.2-{{ .ARCH_SUFFIX }}.raw
+    hard: false
+`
+
 var plog = capnslog.NewPackageLogger("github.com/flatcar/mantle", "kola/tests/misc")
 
 func init() {
 	register.Register(&register.Test{
-		Name:        "cl.misc.nvidia",
-		Run:         verifyNvidiaInstallation,
-		ClusterSize: 1,
-		Distros:     []string{"cl"},
-		// This test is to test the NVIDIA installation, limited to AZURE for now
-		Platforms:     []string{"azure"},
-		Architectures: []string{"amd64"},
+		Name:          "cl.misc.nvidia",
+		Run:           verifyNvidiaInstallation,
+		ClusterSize:   0,
+		Distros:       []string{"cl"},
+		Platforms:     []string{"azure", "aws"},
+		Architectures: []string{"amd64", "arm64"},
 		Flags:         []register.Flag{register.NoEnableSelinux},
+		SkipFunc:      skipOnNonGpu,
+	})
+
+	register.Register(&register.Test{
+		Name:          "cl.misc.nvidia-operator",
+		Run:           verifyNvidiaGpuOperator,
+		ClusterSize:   0,
+		Distros:       []string{"cl"},
+		Platforms:     []string{"azure", "aws"},
+		Architectures: []string{"amd64", "arm64"},
+		Flags:         []register.Flag{register.NoEnableSelinux, register.NoEmergencyShellCheck},
+		SkipFunc:      skipOnNonGpu,
 	})
 }
 
-func verifyNvidiaInstallation(c cluster.TestCluster) {
-	if kola.AzureOptions.Size != "Standard_NC6s_v3" {
-		c.Skip("skipping due to wrong instance size")
+func skipOnNonGpu(version semver.Version, channel, arch, platform string) bool {
+	// N stands for GPU instance obviously :)
+	if platform == "azure" && strings.Contains(kola.AzureOptions.Size, "N") {
+		return false
+	}
+	if platform == "aws" && (strings.HasPrefix(kola.AWSOptions.InstanceType, "p") || strings.HasPrefix(kola.AWSOptions.InstanceType, "g")) {
+		return false
 	}
-	m := c.Machines()[0]
+	return true
+}
 
+func waitForNvidiaDriver(c *cluster.TestCluster, m *platform.Machine) error {
 	nvidiaStatusRetry := func() error {
-		out, err := c.SSH(m, "systemctl status nvidia.service")
+		out, err := c.SSH(*m, "systemctl status nvidia.service")
 		if !bytes.Contains(out, []byte("active (exited)")) {
 			return fmt.Errorf("nvidia.service: %q: %v", out, err)
 		}
 		return nil
 	}
 
 	if err := util.Retry(40, 15*time.Second, nvidiaStatusRetry); err != nil {
+		return err
+	}
+	return nil
+}
+
+func verifyNvidiaInstallation(c cluster.TestCluster) {
+	params := map[string]string{}
+	// Earlier driver versions have issue building on arm64 with kernel 6.6
+	if kola.QEMUOptions.Board == "arm64-usr" {
+		params["NVIDIA_DRIVER_VERSION_LINE"] = "NVIDIA_DRIVER_VERSION=570.86.15"
+	} else {
+		params["NVIDIA_DRIVER_VERSION_LINE"] = ""
+	}
```

Comment:
Would this still be necessary if we updated the nvidia-drivers package in overlay?

Reply:
nope, done